custom_components/miele/sensor.py

```diff
@@ -627,18 +627,25 @@ def state(self):
         device_state = self._device["state"]
         device_status_value = self._device["state"]["status"]["value_raw"]
 
-        if not _is_running(device_status_value):
-            self._cached_consumption = -1
-            return 0
-
         if self._cached_consumption >= 0:
             if (
                 "ecoFeedback" not in device_state
                 or device_state["ecoFeedback"] is None
                 or device_status_value == STATUS_NOT_CONNECTED
             ):
+                # Sometimes the Miele API seems to return a null ecoFeedback
+                # object even though the Miele device is running. Or if the the
+                # Miele device has lost the connection to the Miele cloud, the
+                # status is "not connected". Either way, we need to return the
+                # last known value until the API starts returning something
+                # sane again, otherwise the statistics generated from this
+                # sensor would be messed up.
```

Comment:
I restored this comment too, I don't know if it was removed by mistake or intentionally @kloknibor? I think it provides some value of why this needs to be done.